src/entries/popup/hooks/useTransactionListForPendingTxs.ts

```diff
@@ -106,15 +112,60 @@ function watchForPendingTransactionsReportedByRainbowBackend({
     }
   }
 
-  const updatedPendingTransactions = pendingTransactions?.filter((tx) => {
+  let newlyConfirmedTransactions: RainbowTransaction[] = [];
+  let updatedPendingTransactions: RainbowTransaction[] = [];
+
+  pendingTransactions.forEach((tx) => {
     const txNonce = tx.nonce || 0;
     const latestTx = latestTransactions.get(tx.chainId);
     const latestTxNonce = latestTx?.nonce || 0;
     // still pending or backend is not returning confirmation yet
     // if !latestTx means that is the first tx of the wallet
-    return !latestTx || txNonce > latestTxNonce;
+    const newlyConfirmed = latestTxNonce && txNonce <= latestTxNonce;
+    if (newlyConfirmed) {
+      newlyConfirmedTransactions = [...newlyConfirmedTransactions, tx];
```

Comment:
took me some time to follow this, why not `.push` so `newlyConfirmedTransactions` can be const ?

Reply:
alright, done